libraries/AP_Mount/AP_Mount_Backend.cpp

```diff
@@ -118,6 +118,34 @@ void AP_Mount_Backend::set_rate_target(float roll_degs, float pitch_degs, float
     }
 }
 
+void AP_Mount_Backend::control_mount_open_servo() {
+    // check if the open function is assigned to a servo channel.
+    uint8_t open_chan;
+    if (!SRV_Channels::find_channel(SRV_Channel::k_mount_open, open_chan) &&
+        !SRV_Channels::find_channel(SRV_Channel::k_mount2_open, open_chan)) {
+        return; // open function not assigned, do nothing.
+    }
+
+    // get the configured PWM values for open and closed states.
+    SRV_Channel *channel = SRV_Channels::srv_channel(open_chan);
+    if (!channel) {
+        return; // channel not found.
+    }
+    uint16_t pwm_open = channel->get_output_max();
+    uint16_t pwm_closed = channel->get_output_min();
+
+    // determine the desired open state based on the current mode.
+    bool mount_open = true;
+    if (get_mode() == MAV_MOUNT_MODE_RETRACT) {
+        mount_open = false;
+    }
+
+    // move the servo to the appropriate position.
+    uint16_t target_pwm = mount_open ? pwm_open : pwm_closed;
+    SRV_Channels::move_servo(SRV_Channel::Aux_servo_function_t(open_chan), 
+                              target_pwm, pwm_closed, pwm_open);
```

Comment:
```suggestion
    // move the servo to the appropriate position.
    SRV_Channels::set_output_scaled(SRV_Channel::Aux_servo_function_t(open_chan), mount_open ? 1.0 : 0.0; );
```
Most of the function can be replaced with a set scaled. No need to get the individual output channel and mess about with PWM. 

For the set scaled to work you just need to set the servo to a range output in the init somewhere:
```
SRV_Channels::set_range(SRV_Channel::k_mount_open, 1.0);
SRV_Channels::set_range(SRV_Channel::k_mount2_open, 1.0);
```